### PR TITLE
Live Location Sharing - Background permission

### LINKED ISSUE #5536: [Location sharing] - Live location permission dialog
#### What would you like to do?
Make the display of live location sharing option depending on a Build config.
When choosing live location sharing option, show a dialog if location permission is not in "Always" state.

#### Why would you like to do it?
To share location as a live, location permission should be in the "Always" state.

#### How would you like to achieve it?
See https://www.figma.com/file/uthEK6xuo9hSQRdWvPAre5/Location-Sharing?node-id=378%3A50863

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by onurays**
Fixes #5536.

Background location permission behavior started to vary with API 29 as follows:

**"<10"** -> background location permission is automatically granted by the system
**"=10"** -> The system permission dialog includes "Allow all the time" to grant the background location permission
**">10"** -> The system permission dialog DOES NOT include "Allow all time". Users need to enable it in settings

### CODE REVIEW

**Review by mnaturel — COMMENTED** [review]
I have just small improvement suggestions we could do and some tests to do on devices.

File: vector/src/main/java/im/vector/app/features/location/DefaultLocationSharingSettingsNavigator.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    override var goingToAppSettings: Boolean = false
+
+    override fun quit() {
+        activity?.finish()
+    }
+
+    override fun goToAppSettings() {
+        activity?.let {
+            goingToAppSettings = true
+            openAppSettingsPage(it)
```
[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/DefaultLocationSharingSettingsNavigator.kt:33]: In fact `ActivityResultLauncher` does not work as I expected on Android 10 for example. It requests permission as usual with dialog inside the app. So let's keep the current implementation.

File: vector/src/main/java/im/vector/app/features/location/DefaultLocationSharingSettingsNavigator.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location
+
+import android.app.Activity
+import im.vector.app.core.utils.openAppSettingsPage
+
+class DefaultLocationSharingSettingsNavigator constructor(val activity: Activity?) : LocationSharingSettingsNavigator {
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/DefaultLocationSharingSettingsNavigator.kt]: Right, done.

File: vector/src/main/java/im/vector/app/features/location/LocationSharingSettingsNavigator.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location
+
+interface LocationSharingSettingsNavigator {
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/location/LocationSharingSettingsNavigator.kt]: Done.

**Review by mnaturel — APPROVED** [review]

**Pull request merged**